swap/src/asb/rendezvous.rs

```diff
@@ -0,0 +1,155 @@
+                if Instant::now() > rendezvous_reregister_timestamp {
+                    self.rendezvous_behaviour.register(
+                        Namespace::new(self.rendezvous_namespace.to_string())
+                            .expect("our namespace to be a correct string"),
+                        self.rendezvous_point_peer_id,
+                        None,
+                    )?;
+                }
+            }
+        } else {
+            let p2p_suffix = Protocol::P2p(self.rendezvous_point_peer_id.into());
+            let address_with_p2p = if !self
+                .rendezvous_point_addr
+                .ends_with(&Multiaddr::empty().with(p2p_suffix.clone()))
+            {
+                self.rendezvous_point_addr.clone().with(p2p_suffix)
+            } else {
+                self.rendezvous_point_addr.clone()
+            };
+            self.events.push(NetworkBehaviourAction::DialAddress {
+                address: address_with_p2p,
+            })
+        }
+        Ok(())
+    }
+}
+
+type BehaviourInEvent =
+<<<libp2p::rendezvous::Rendezvous as NetworkBehaviour>::ProtocolsHandler as IntoProtocolsHandler>::Handler as ProtocolsHandler>::InEvent;
+
+impl NetworkBehaviour for Behaviour {
+    type ProtocolsHandler = <libp2p::rendezvous::Rendezvous as NetworkBehaviour>::ProtocolsHandler;
+    type OutEvent = ();
+
+    fn new_handler(&mut self) -> Self::ProtocolsHandler {
+        <libp2p::rendezvous::Rendezvous as NetworkBehaviour>::ProtocolsHandler::default()
+    }
+
+    fn addresses_of_peer(&mut self, _peer_id: &PeerId) -> Vec<Multiaddr> {
+        vec![]
+    }
+
+    fn inject_connected(&mut self, peer_id: &PeerId) {
+        if *peer_id == self.rendezvous_point_peer_id {
+            self.is_connected = true;
+        }
+    }
+
+    fn inject_disconnected(&mut self, peer_id: &PeerId) {
+        if *peer_id == self.rendezvous_point_peer_id {
+            self.is_connected = false;
+        }
+    }
+
+    fn inject_event(
+        &mut self,
+        _peer_id: PeerId,
+        _connection: ConnectionId,
+        _event: <<libp2p::rendezvous::Rendezvous as NetworkBehaviour>::ProtocolsHandler as ProtocolsHandler>::OutEvent,
+    ) {
+    }
+
+    fn poll(
+        &mut self,
+        _cx: &mut Context<'_>,
+        _params: &mut impl PollParameters,
+    ) -> Poll<NetworkBehaviourAction<BehaviourInEvent, Self::OutEvent>> {
+        if let Some(event) = self.events.pop() {
+            return Poll::Ready(event);
+        }
+        Poll::Pending
+    }
+}
+
+impl NetworkBehaviourEventProcess<libp2p::rendezvous::Event> for Behaviour {
+    fn inject_event(&mut self, event: Event) {
+        match event {
+            Event::RegisterFailed(error) => {
+                tracing::error!(rendezvous_node=%self.rendezvous_point_peer_id, "Registration with rendezvous node failed: {:#}", error);
+            }
+            Event::RegistrationExpired(registration) => {
```

Comment:
Why was this re-added? As I stated before, this is only relevant for the rendezvous node, not for the registration side. Any particular reason why it was added?